### ISSUE TITLE
How does routing in Nebula work?

### ISSUE THREAD

**Issue by drudru** [slackhq/nebula#368]
Nebula looks really neat.

How are packets routed through the nebula overlay?

**Comment by zeisss** [comment]
Currently only directly between the involved nodes. The lighthouse is only used to find other nodes.

Afaik there is some thinking to reroute packets via other nodes in the works.

**Comment by drudru** [comment]
So, the system is pretty flat then?
What happens if two nodes cannot directly communicate with each other?

**Comment by zeisss** [comment]
Yes. There is some logic for hole-punching between two nodes behind NAT. 

Checkout [this talk from November](https://www.youtube.com/watch?v=qy2cgqglt3o) from @rawdigits which might be of interest explaining how things work.

**Comment by drudru** [comment]
Thank you @zeisss. That is a very long video. I will check it out during a weekend.

**Comment by zeisss** [comment]
@drudru I am unaware of it but I am also not involved in the Nebula development either. Maybe this is something you can bring up in the Nebula Slack.
